azurerm/internal/common/correlation_id.go

```diff
@@ -24,12 +26,17 @@ func withCorrelationRequestID(uuid string) autorest.PrepareDecorator {
 	return autorest.WithHeader(HeaderCorrelationRequestID, uuid)
 }
 
-// correlationRequestID generates an UUID to pass through `x-ms-correlation-request-id` header.
+// correlationRequestID either uses a user provided correlation request id or generates a UUID to pass through `x-ms-correlation-request-id` header.
 func correlationRequestID() string {
 	msCorrelationRequestIDOnce.Do(func() {
+		msCorrelationRequestID = os.Getenv(envArmCorrelationRequestID)
```

Comment:
rather than pulling this from an env var - we should instead make this configurable via the provider block, so let's remove this:

```suggestion
```

meaning we could instead add something like this:

```
	if !o.DisableCorrelationRequestID {
		if o.CustomCorrelationRequestID != nil {
			useCustomCorrelationRequestID(*o.CustomCorrelationRequestID)
		}
		c.RequestInspector = withCorrelationRequestID(correlationRequestID())
	}
```

here: https://github.com/terraform-providers/terraform-provider-azurerm/blob/5facd461e7075074c8270d68d884692c201f7c45/azurerm/internal/common/client_options.go#L44-L46

and presumably add a setter method in the form of `useCustomCorrelationRequestID` to this file, which'd allow this to be configured in the provider block and via an env var - WDYT?

Reply:
yeah sorry I'd meant to edit this but forgot 🤦 - we can add this to the `ClientBuilder` struct but still feed this from an env var, it just means it's pulled from a single place, which achieves the same thing (e.g. having all of those in the same place, so this isn't hidden) but means this isn't exposed in that block